src/yggdrasil/adapter.go

```diff
@@ -1,18 +1,47 @@
 package yggdrasil
 
-// Defines the minimum required struct members for an adapter type (this is
-// now the base type for tunAdapter in tun.go)
+import (
+	"github.com/gologme/log"
+	"github.com/yggdrasil-network/yggdrasil-go/src/address"
+	"github.com/yggdrasil-network/yggdrasil-go/src/config"
+)
+
+// Adapter defines the minimum required struct members for an adapter type. This
+// is now the base type for adapters like tun.go. When implementing a new
+// adapter type, you should extend the adapter struct with this one and should
+// call the Adapter.Init() function when initialising.
 type Adapter struct {
```

Comment:
General reminder that I'm not really happy with this interface and want to do something else eventually.

For example, we might want to have `src/tuntap` (or some utility library that it uses) figure out the NodeID to use based on the packet's , then do something like a `Dial()` but with Yggdrasil's NodeIDs, and get back something resembling a `net.Conn`, or maybe skip the `Dial()` step and just expose a `net.PacketConn`. The latter may be easier, but keeps a lot of session management logic in `src/yggdrasil` that may make more sense elsewhere. The point is, ideally we should try to conform to `net`'s interfaces, so we can be a drop-in replacement for other things.

Reply:
Yep – I'd also like to get to the point where `Adapter` is not really necessary, but for now it solved a short-term problem just to keep Yggdrasil buildable and usable with this PR.